indico/modules/events/papers/models/revisions.py

```diff
@@ -152,3 +158,27 @@ def paper(self):
     @paper.setter
     def paper(self, paper):
         self._contribution = paper.contribution
+
+    def get_timeline(self, user=None):
+        comments = [x for x in self.comments if x.can_view(user)] if user else self.comments
+        reviews = [x for x in self.reviews if x.can_view(user)] if user else self.reviews
+        return sorted(chain(comments, reviews), key=attrgetter('created_dt'))
+
+    def get_reviews(self, group=None, user=None):
+        reviews = []
+        if user and group:
+            reviews = [x for x in self.reviews if x.group.instance == group and x.user == user]
```

Comment:
because I think it looks better this way! I considered that too, but I prefer 1 return statement rather than 4, makes it cleaner imo.